controllers/version_update.go

```diff
@@ -154,6 +157,16 @@ func upgradeForBackend(master *v1alpha1.CDAPMaster, labels map[string]string, ob
 		return jobObj
 	}
 
+	// Skip pre-upgrade and post-upgrade jobs for patch revisions
```

Comment:
All the logic to check whether Pre Upgrade job should run should be done at the start. Line  127.